hooks/pre-commit

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+if clang-format --version | grep -q 'version 11\.'; then
+   CLANG_FORMAT_EXECUTABLE="clang-format"
+else
+   CLANG_FORMAT_EXECUTABLE="clang-format-11"
+fi
+
+if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
+then
+    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
+    exit 1
+fi
+
+for FILE in $(git diff --cached --name-only)
+do
+	if [[ "$FILE" =~ src/[A-Za-z0-9\ \-]+*\.(c|h|cpp|cc)$ ]]; then
+		echo Autoformatting $FILE with $CLANG_FORMAT_EXECUTABLE
+		$CLANG_FORMAT_EXECUTABLE -style=file -i -- $FILE
+		git add -- $FILE
+	elif [[ "$FILE" =~ src/(components|displayapp|drivers|heartratetask|logging|systemtask)/.*\.(c|h|cpp|cc)$ ]]; then
+		echo Autoformatting $FILE with $CLANG_FORMAT_EXECUTABLE
+		$CLANG_FORMAT_EXECUTABLE -style=file -i -- $FILE
+		git add -- $FILE
```

Comment:
These are quite hard to read. It seems that the top one matches all files in src and the bottom one ignores libs and FreeRTOS folders. Would it make more sense to specifically ignore these
```
if [[ "$FILE" =~ src/libs/.* ]] || [[  "$FILE" =~ src/FreeRTOS/.* ]]
then
  continue
fi
```
and match everything else based on just the extension `if [[ "$FILE" =~ .*\.(c|h|cpp|cc)$ ]]`

Reply:
It's intentionally an explicit and a predefined choice to avoid any unnecessary formats.

Have you tested your suggestion?

Reply:
I tested it separately and now in the repo and it works fine.